core/client/src/main/java/alluxio/client/AbstractOutStream.java

```diff
@@ -28,12 +28,20 @@
   // requires checking the codebase for this assumption (using int to denote an offset).
   // See ALLUXIO-1536.
   /** The number of bytes written. */
-  protected int mBytesWritten = 0;
+  protected long mBytesWritten = 0;
 
   /**
    * @return the number of bytes written to this stream
+   * @deprecated use {@link #getTotalBytesWritten()} to avoid integer overflow
    */
+  @Deprecated
   public int getBytesWritten() {
+    // Cast to int to preserve backwards compatibility. In 2.0 we should change this to long so that
+    // it can be correct.
+    return (int) mBytesWritten;
+  }
```

Comment:
I don't care too much what manner we're incorrect in, but as a general rule prefer a more obviously incorrect behavior over a less obviously incorrect one. If we has used MAX_INT here, it would have been harder to discover that we were overflowing.